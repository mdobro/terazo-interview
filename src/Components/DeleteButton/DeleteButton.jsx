import React, { useState } from "react";
import PropTypes from "prop-types";
import { Modal, ModalBody, ModalFooter, ModalHeader, Button } from "reactstrap";

const DeleteButton = ({ onDelete, modalHeader, modalBody }) => {
  const [openModal, setOpenModal] = useState(false);

  return (
    <>
      <Modal data-testid="deleteModal" isOpen={openModal}>
        <ModalHeader>{modalHeader}</ModalHeader>
        <ModalBody>{modalBody}</ModalBody>
        <ModalFooter>
          <Button
            className="deleteModalButton"
            data-testid="deleteModalButton"
            color="danger"
            onClick={onDelete}
          >
            Delete
          </Button>{" "}
          <Button onClick={() => setOpenModal(false)}>Cancel</Button>
        </ModalFooter>
      </Modal>
      <Button
        data-testid="deleteButton"
        color="danger"
        size="sm"
        onClick={() => setOpenModal(true)}
      >
        Delete
      </Button>
    </>
  );
};

DeleteButton.propTypes = {
  onDelete: PropTypes.func.isRequired,
  modalHeader: PropTypes.string,
  modalBody: PropTypes.string,
};

DeleteButton.defaultProps = {
  modalHeader: "Are you sure?",
  modalBody:
    "Deleting this will remove it from the system. This cannot be undone.",
};

export default DeleteButton;
