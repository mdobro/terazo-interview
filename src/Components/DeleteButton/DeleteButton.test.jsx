import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import DeleteButton from "./DeleteButton";

test("renders", () => {
  const { asFragment } = render(<DeleteButton />);
  expect(asFragment()).toMatchSnapshot();
});

test("Shows modal", () => {
  const onDelete = jest.fn();
  const { getByTestId, queryByTestId } = render(
    <DeleteButton onDelete={onDelete} />
  );
  expect(queryByTestId("deleteModal")).toBeFalsy();
  const deleteButton = getByTestId("deleteButton");
  userEvent.click(deleteButton);
  expect(queryByTestId("deleteModal")).toBeTruthy();
});

test("Calls onDelete", () => {
  const onDelete = jest.fn();
  const { getByTestId, queryByTestId } = render(
    <DeleteButton onDelete={onDelete} />
  );
  const deleteButton = getByTestId("deleteButton");
  userEvent.click(deleteButton);
  const modalDeleteButton = getByTestId("deleteModalButton");
  userEvent.click(modalDeleteButton);
  expect(onDelete).toHaveBeenCalled();
});
