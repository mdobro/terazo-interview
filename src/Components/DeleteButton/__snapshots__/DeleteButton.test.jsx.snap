// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    class="btn btn-danger btn-sm"
    data-testid="deleteButton"
    type="button"
  >
    Delete
  </button>
</DocumentFragment>
`;
